TESTS/psa/entropy_inject/main.cpp

```diff
@@ -90,6 +102,9 @@ utest::v1::status_t greentea_test_setup(const size_t number_of_cases)
         seed[i] = i;
     }
 
+    psa_status_t status = mbed_psa_reboot_and_request_new_security_state(PSA_LIFECYCLE_ASSEMBLY_AND_TEST);
+    TEST_ASSERT_EQUAL(PSA_SUCCESS, status);
```

Comment:
This is a new test, not a refactor. Could you add it in a new commit?

Also, does this work? The function says "reboot", but if it actually reboots, what guarantee do we have that this test will resume after this function in the correct place?

Reply:
mbed_psa_reboot_and_request_new_security_state() API replaced its_reset() which is now a secure API only

This change is necessary for a clean environment for the test

Reply:
OK. We would benefit greatly from that being written down in the commit message that makes this change, a separate commit from the refactoring.

Does "reboot" actually reboot, like a soft-power reset?